docs/Formula-Android.md

```diff
@@ -261,30 +261,113 @@ class MyActivity : FragmentActivity() {
 ```
 This is already in place for you if you use `FormulaAppCompatActivity`.
 
-## Fragment flow
-Flow enables to group multiple fragments and share state between all of them. 
+## Fragment flow factory
+A flow factory enables to group multiple fragments and share state, routers, action handlers
+and other dependencies between them. It has two generic parameters:
 
+- `Dependencies` that the parent needs to provide. 
+- `FlowComponent` that will be shared with all features defined by this flow.
+
+For this example, let's say we are building authentication for our app. The backend provides view
+information such as the hints we should show for email or password fields. We want to only fetch 
+this data once and share it between the login and sign up fragments. Let's say our repository 
+class looks something like this
+
+```kotlin
+class AuthRepo(private val retrofit: Retrofit) {
+
+    // Backend provides data that helps us render auth pages.
+    data class PageResponse(
+        val emailHint: String,
+        val passwordHint: String,
+    )
+
+    // This will check if we have in-memory response and
+    // return that otherwise make a fresh call.
+    fun pageEvents(): Observable<AuthPageResponse> { ... }
+}
+```
+
+As you can see above, this repository class needs a `Retrofit` instance to function. We can define
+the dependencies we need using an interface.
+```kotlin
+interface Dependencies {
+    fun retrofit(): Retrofit    
+}
+```
+
+Let's create our flow component. In a real application, this would usually be a `Dagger` component.
```

Comment:
Should we explicitly mention `Dagger` here? Maybe use more tolerant `DI Framework` term?

Reply:
Updating to 
```markdown
In a real application, component would usually be implemented by DI framework such as `Dagger`.
```